src/plugins/mer/mervirtualmachinesettingswidget.cpp

```diff
@@ -0,0 +1,108 @@
+#include "mervirtualmachinesettingswidget.h"
+#include "mervirtualboxmanager.h"
+#include <utils/utilsicons.h>
+#include "ui_mervirtualmachinesettingswidget.h"
+
+namespace Mer {
+namespace Internal {
+
+namespace {
+const int DEFAULT_MAX_MEMORY_SIZE_MB = 3584;
+const int DEFAULT_MAX_VDI_SIZE_GB = 2000;
+
+} // namespace anonymous
+
+MerVirtualMachineSettingsWidget::MerVirtualMachineSettingsWidget(QWidget *parent)
+    : QWidget(parent)
+    , ui(new Ui::MerVirtualMachineSettingsWidget)
+{
+    ui->setupUi(this);
+    initGui();
+}
+
+MerVirtualMachineSettingsWidget::~MerVirtualMachineSettingsWidget()
+{
+    delete ui;
+}
+
+void MerVirtualMachineSettingsWidget::setMemorySizeMb(int sizeMb)
+{
+    if (ui->memorySpinBox->maximum() < sizeMb)
+        ui->memorySpinBox->setRange(4, sizeMb);
+
+    ui->memorySpinBox->setValue(sizeMb);
+}
+
+void MerVirtualMachineSettingsWidget::setCpuCount(int count)
+{
+    ui->cpuCountSpinBox->setValue(count);
+}
+
+void MerVirtualMachineSettingsWidget::setVdiCapacityMb(int vdiCapacityMb)
+{
+    const int vdiCapacityGb = static_cast<int>(vdiCapacityMb / 1024.0);
+    // Prohibit adding more than 10GB in one step, because there is no way to go back
+    int maxVdiSize = vdiCapacityGb + 10 > DEFAULT_MAX_VDI_SIZE_GB ? DEFAULT_MAX_VDI_SIZE_GB : vdiCapacityGb + 10;
+    // the minimum size is the current size because VBoxManager can't reduce storage size
+    ui->vdiCapacityGbSpinBox->setRange(vdiCapacityGb, maxVdiSize);
```

Comment:
Wouldn't this be enough?

```
ui->vdiCapacityGbSpinBox->setRange(vdiCapacityGb, vdiCapacityGb + MAX_VDI_SIZE_INCREMENT_GB);
```

IIUC one would have to increase the size approximately 200 times to get over DEFAULT_MAX_VDI_SIZE_GB the regular way :)